### ISSUE TITLE
Using URL anchors in markdown docs repo's vs 'user-content' prefix.

### ISSUE THREAD

**Issue by aschrijver** [go-gitea/gitea#13748]
- Gitea version (or commit ref): 1.12.3 ([Codeberg](https://codeberg.org))

## Description

When using anchors within the same Markdown doc, then upon save everything works fine. The anchors are processed and the prefix `user-content-` is added.

But when referencing an anchor in another Markdown document in the same repo, then this doesn't work, and the prefix must be manually added to the anchor text. E.g.

```markdown
- [Header 1 works](#header-1-works)
- [Subpage Header 1 does not](subfolder/subpage.md#subpage-header-1-does-not)
- [Subpage Header 1 prefixed does](subfolder/subpage.md#user-content-subpage-header-1-prefixed-does)

And <a href="subfolder/subpage.md#subpage-header-1">does not work either.</a>

# Header 1 works
```
Maybe it is hard to change this behavior, as it would require more parsing and checks of the referenced page, probably. But for working in a documentation repository with potentially many pages and cross-references (I intend to follow [readme-driven development](https://tom.preston-werner.com/2010/08/23/readme-driven-development.html)) this will be very confusing for all the people involved in refining said docs.

**Comment by silverwind** [comment]
Please show an example on https://try.gitea.io.

**Comment by aschrijver** [comment]
You can see the issue fixes in these commits on the main README: [`551e76518f`](https://codeberg.org/teaserbot-labs/outreach-matters/commit/551e76518f863629c5b737c763a3c3f2dd4b8cdf), [`56da6688c9`](https://codeberg.org/teaserbot-labs/outreach-matters/commit/56da6688c974ec813d758df68d8d5d8fe13034db)

**Comment by GuySartorelli** [comment]
This looks to have been fixed in 1.13.0 (see #12062)

**Comment by lunny** [comment]
Closed as fixed. Please feel free to reopen it.
